examples/mysql-galera/image/docker-entrypoint.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+HOSTNAME=`hostname`
+if [ -z "$NUM_NODES" ]; then
+  NUM_NODES=3
+fi
+
+if [ "${1:0:1}" = '-' ]; then
+  set -- mysqld "$@"
+fi
+
+if [ "$1" = 'mysqld' ]; then
+  # read DATADIR from the MySQL config
+  DATADIR="$("$@" --verbose --help 2>/dev/null | awk '$1 == "datadir" { print $2; exit }')"
+  
+  if [ ! -d "$DATADIR/mysql" ]; then
+    if [ -z "$MYSQL_ROOT_PASSWORD" -a -z "$MYSQL_ALLOW_EMPTY_PASSWORD" ]; then
+      echo >&2 'error: database is uninitialized and MYSQL_ROOT_PASSWORD not set'
+      echo >&2 '  Did you forget to add -e MYSQL_ROOT_PASSWORD=... ?'
+      exit 1
+    fi
+
+    echo 'Running mysql_install_db ...'
+        mysql_install_db --datadir="$DATADIR"
+        echo 'Finished mysql_install_db'
+
+    
+    # These statements _must_ be on individual lines, and _must_ end with
+    # semicolons (no line breaks or comments are permitted).
+    # TODO proper SQL escaping on ALL the things D:
+    
+    tempSqlFile='/tmp/mysql-first-time.sql'
+    cat > "$tempSqlFile" <<-EOSQL
+DELETE FROM mysql.user ;
+CREATE USER 'root'@'%' IDENTIFIED BY '${MYSQL_ROOT_PASSWORD}' ;
+GRANT ALL ON *.* TO 'root'@'%' WITH GRANT OPTION ;
+DROP DATABASE IF EXISTS test ;
```

Comment:
I removed the drop table test but the rest of it is necessary run at the end: 

set -- "$@" --init-file="$tempSqlFile". 

MySQL starts up initially with that, then restarts again after all that is set so that permissions are there for the root account as well as SST user, etc.